src/client/actions/UserUpdate.js

```diff
@@ -1,22 +1,33 @@
 const Action = require('./Action');
 const { Events } = require('../../util/Constants');
+const User = require('../../structures/User');
 
 class UserUpdateAction extends Action {
   handle(data) {
     const client = this.client;
 
+    const newUser = new User(client, data.user);
```

Comment:
Wouldn't this break references?
I think this is more of a use case for `Base#_update`.

I am also not sure how this is supposed to play well with updates of the ClientUser, as this would insert that new User (not ClientUser) into the cache then.